File: LeetCode/Problems/Python/1099. Two Sum Less Than K.py
class Solution:
    def twoSumLessThanK(self, A: List[int], K: int) -> int:
        A.sort()
        lo = 0
        hi = len(A)-1
        
        ans = -1
        while lo < hi:
            sum = A[lo] + A[hi]
            if sum >= K:
                hi-=1
            else:
                ans = max(ans, sum)
                lo+=1
        return ans

File: LeetCode/Problems/C++/1099. Two Sum Less Than K.cpp
//Two pointers
//Time: O(n)
//Space: O(1) 
class Solution {
public:
    int twoSumLessThanK(vector<int>& A, int K) {
        sort(A.begin(), A.end());
        
        int lo = 0, hi = A.size()-1;
        int minSum = -1, sum;      
        while(lo<hi){
            sum = A[lo] + A[hi];
            if(sum >= K){
                hi--;
            }
            else{
                minSum = max(minSum, sum);
                lo++;
            }
        }
        return minSum;
    }
};